src/debugAdapter/goDebug.ts

```diff
@@ -1587,6 +1585,78 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	private logDelveError(err: any, message: string) {
+		if (err === undefined) {
+			return;
+		}
+
+		let errorMessage = err.toString();
+		// Handle unpropagated fatalpanic errors with a more user friendly message:
+		// https://github.com/microsoft/vscode-go/issues/1903#issuecomment-460126884
+		// https://github.com/go-delve/delve/issues/852
+		// This affects macOS only although we're agnostic of the OS at this stage, only handle the error
+		if (errorMessage === 'bad access') {
+			errorMessage = 'unpropagated fatalpanic: signal SIGSEGV (EXC_BAD_ACCESS). This fatalpanic is not traceable on macOS, see https://github.com/go-delve/delve/issues/852';
+		}
+
+		logError(message + ' - ' + errorMessage);
+
+		if (err.toString() === 'bad access') {
+			logError('WARNING: this stack might not be from the expected active goroutine');
+		}
+
+		this.dumpStacktrace();
+	}
+
+	private dumpStacktrace() {
+		// Get current goroutine
+		// Debugger may be stopped at this point but we still can (and need) to obtain state and stacktrace
+		let goroutineId = 0;
+		this.delve.call<DebuggerState | CommandOut>('State', [{ NonBlocking: true }], (err, out) => {
+			if (!err) {
+				goroutineId = this.delve.isApiV1 ? (<CommandOut>out).State.currentGoroutine.id : (<DebuggerState>out).currentGoroutine.id;
+			}
+		});
+
+		// Get goroutine stacktrace
+		const stackTraceIn = { id: goroutineId, depth: this.delve.stackTraceDepth };
```

Comment:
The `goroutineId` here will always be 0 as this code path is not inside the callback provided to the delve call a few lines above.

Reply:
You can try

```javascript
try {
  const stateCallResult: DebuggerState | CommandOut = await this.delve.callPromise('State', [{ NonBlocking: true }]);
  goroutineId = this.delve.isApiV1 ? (<CommandOut>stateCallResult).State.currentGoroutine.id : (<DebuggerState>stateCallResult).currentGoroutine.id;
} catch (error) {
	// Ignore error?
}
```

Reply:
@jhendrixMSFT @ramya-rao-a I think I solved the issue of state returning always (?) zero. The gist of the debugger state is that there seems to be no currentGoroutine in this scenario so we have to default to the currentThread goroutineId:

This is the apicall response:
```json
{
	"Running": false,
	"currentThread": {
		"id": 5142829,
		"pc": 17138024,
		"file": "/Users/lgomez/DEV/go/test_1903/main.go",
		"line": 5,
		"function": {
			"name": "main.oops",
			"value": 17138000,
			"type": 0,
			"goType": 0,
			"optimized": false
		},
		"goroutineID": 1,
		"ReturnValues": null
	},
	"Threads": [
		{
			"id": 5142829,
			"pc": 17138024,
			"file": "/Users/lgomez/DEV/go/test_1903/main.go",
			"line": 5,
			"function": {
				"name": "main.oops",
				"value": 17138000,
				"type": 0,
				"goType": 0,
				"optimized": false
			},
			"goroutineID": 1,
			"ReturnValues": null
		},
		{
			"id": 5142873,
			"pc": 140734907268914,
			"file": "",
			"line": 0,
			"goroutineID": 0,
			"ReturnValues": null
		},
		{
			"id": 5142874,
			"pc": 140734907267178,
			"file": "",
			"line": 0,
			"goroutineID": 0,
			"ReturnValues": null
		},
		{
			"id": 5142875,
			"pc": 140734907267178,
			"file": "",
			"line": 0,
			"goroutineID": 0,
			"ReturnValues": null
		},
		{
			"id": 5142876,
			"pc": 140734907267178,
			"file": "",
			"line": 0,
			"goroutineID": 0,
			"ReturnValues": null
		}
	],
	"NextInProgress": false,
	"exited": false,
	"exitStatus": 0,
	"When": ""
}
```
 
And this is the handled panic with the updated behavior:
<img width="777" alt="Screen Shot 2019-12-19 at 17 59 35" src="https://user-images.githubusercontent.com/6326271/71209205-a60daf00-2289-11ea-9099-f277efba1d25.png">

This is on mac, I´ll try this at home later to see if the response is different on linux

Reply:
I'm not getting the currentGoroutine on linux either, i'm trying other goroutine examples and i'm even reproducing #2563 while at it so I'm willing to say this is our best bet for now. We're supposed to be getting this field but we can follow up on that issue since it is better suited for its root cause